share/spack/csh/pathadd.csh

```diff
@@ -25,8 +25,8 @@ if ($_pa_set == 1) then
     eval set _pa_old_value='$'$_pa_varname
 endif
 
-# Do the actual prepending here, if it is a dir and not already in the path
-if ( -d $_pa_new_path && \:$_pa_old_value\: !~ *\:$_pa_new_path\:* ) then
+# Do the actual prepending here, if it is a dir and not first in the path
+if ( -d $_pa_new_path && $_pa_old_value\: !~ $_pa_new_path\:* ) then
```

Comment:
Part of the prefixed `:` is to make sure we don't match with partial paths. If I want to add `/b/c` and `/a/b/c` is already there, won't this now also indicate a match? Might need an explicit `^`?

Reply:
I am no csh expert either, but I tested and found that csh matches the entire string; so there is an implicit `^` here I think.

For example, this does *not* "match":
```csh
if ( "abtest" =~ te* ) then
    echo match
endif
```

Reply:
Actually now I can recall adding `^` doesn't even work: I don't think this is a regex, but a csh-specific pattern.

Reply:
Yeah, apparently just a glob pattern.